packages/@react-spectrum/s2/src/SelectBox.tsx

```diff
@@ -0,0 +1,308 @@
+    orientation: {
+      horizontal: {
+        size: {
+          S: size(312),
+          M: size(368)
+        }
+      },
+      vertical: {
+        size: {
+          XS: size(168),
+          S: size(184),
+          M: size(200),
+          L: size(216),
+          XL: size(232)
+        }
+      }
+    }
+  },
+  minWidth: {
+    orientation: {
+      horizontal: {
+        size: {
+          S: size(312),
+          M: size(368)
+        }
+      },
+      vertical: {
+        size: {
+          XS: size(100),
+          S: size(128),
+          M: size(144),
+          L: size(160),
+          XL: size(192)
+        }
+      }
+    }
+  },
+  outlineColor: {
+    default: 'focus-ring',
+    forcedColors: 'Highlight'
+  },
+  padding: {
+    orientation: {
+      horizontal: {
+        default: 16
+      }
+    }
+  },
+  position: 'relative'
+});
+
+const selectBoxContentStyle = style({
+  alignContent: 'center',
+  alignItems: 'center',
+  display: 'grid',
+  gap: {
+    orientation: {
+      horizontal: {
+        size: {
+          S: 0,
+          M: size(2)
+        },
+        vertical: size(8)
+      }
+    }
+  },
+  gridTemplateAreas: {
+    orientation: {
+      horizontal: ['icon label', 'icon description'],
+      vertical: ['. icon .', '. label .']
+    }
+  },
+  gridTemplateColumns: {
+    orientation: {
+      horizontal: '48px 1fr'
+    }
+  },
+  height: 'full',
+  justifyItems: {
+    orientation: {
+      horizontal: 'flex-start',
+      vertical: 'center'
+    }
+  },
+  paddingStart: {
+    orientation: {
+      horizontal: size(12),
+      vertical: 0
+    }
+  },
+  width: 'auto'
+});
+
+const selectBoxIconStyle = style({
+  color: {
+    default: 'red-600',
+    isDisabled: 'disabled'
+  },
+  gridArea: 'icon',
+  marginBottom: {
+    orientation: {
+      horizontal: 0,
+      vertical: 8
+    }
+  },
+  flexShrink: 0
+});
+
+const selectBoxLabelStyle = style({
+  gridArea: 'label',
+  color: {
+    default: 'neutral',
+    isDisabled: 'disabled'
+  },
+  fontSize: {
+    size: {
+      XS: 'body-sm',
+      M: 'body-lg',
+      L: 'body-xl',
+      XL: 'body-2xl'
+    }
+  }
+});
+
+const selectorStyle = style({
+  display: 'block',
+  position: 'absolute',
+  top: {
+    orientation: {
+      vertical: 16,
+      horizontal: '50%'
+    }
+  },
+  right: 16,
+  visibility: {
+    default: 'hidden',
+    isHovered: 'visible',
+    isSelected: 'visible'
+  }
+});
+
+const SelectBox = (props: SelectBoxProps, ref: FocusableRef<HTMLLabelElement>) => {
+  const {orientation, selectionMode, size} = useSelectBoxGroupProvider();
+  const AriaSelector = selectionMode === 'single' ? AriaRadio : AriaCheckbox;
+  const Selector = selectionMode === 'single' ? Radio : Checkbox;
+  const domRef = useFocusableRef<HTMLLabelElement>(ref);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  return (
+    <AriaSelector
+      className={renderProps => selectBoxStyle({...renderProps, orientation, size})}
```

Comment:
We need to include UNSAFE classname/style and style overrides, see https://github.com/adobe/react-spectrum/blob/e92ccd7568db87ed15b828b9e0b4a7de5dababd1/packages/%40react-spectrum/s2/src/Card.tsx#L384